load_balancer/test/functional/load_balancerTest.js

```diff
@@ -0,0 +1,449 @@
+    const responseBody = JSON.parse(response.body);
+    responseBody.components.forEach((elem, index) => {
+      if (upArray.includes(index)) {
+        elem.status.should.equal('up');
+      } else {
+        elem.status.should.equal('down');
+      }
+    });
+    restoreConsole();
+    resp();
+  });
+};
+
+const jobSubmitCheck = function jobSubmitCheck() {
+  const tempJob = testData.submissionsData[0];
+  request.post(`${lbUrl}/submit`, { json: tempJob }, (error, response) => {
+    (error === null).should.be.true();
+    response.body.should.equal(true);
+  });
+};
+
+const submitReqCheck = function submitReqCheck(executedJobDetails) {
+  request.post(`${lbUrl}/sendScores`, { json: executedJobDetails }, (error, response) => {
+    (error === null).should.be.true();
+    response.body.should.equal(true);
+  });
+};
+
+const testAssertion = function testAssertion(testDetails, nodeArray, resp) {
+  setTimeout(() => {
+    if (testDetails.requestRunNockObj != null) {
+      testDetails.requestRunNockObj.isDone().should.equal(testDetails.requestRunVal);
+    }
+    if (testDetails.resultsNockObj != null) {
+      testDetails.resultsNockObj.isDone().should.equal(testDetails.resultsVal);
+    }
+    connectionCheckAssert(nodeArray, () => {
+      resp();
+    });
+  }, 100);
+};
+
+describe('Correctly maintains list of ENs', () => {
+  beforeEach(() => {
+    stubConsole();
+    startLoadBalancer();
+    restoreConsole();
+    activateNocks();
+  });
+
+  afterEach((done) => {
+    cleanNocks();
+    loadBalancer.server.close(done);
+  });
+
+  it('during status check', (done) => {
+    stubConsole();
+    const res = JSON.parse(JSON.stringify(configData.Nodes));
+    const lbTestStatus = configData.load_balancer;
+    lbTestStatus.status = 'up';
+    [0, 1].forEach((elem) => {
+      mockENStatus(`https://${res[elem].hostname}:${res[elem].port}`, true);
+      mockENStatus(`https://${res[elem].hostname}:${res[elem].port}`, true);
+    });
+    request.get(`${lbUrl}/connectionCheck`, (error, response) => {
+      const responseBody = JSON.parse(response.body);
+      response.should.exist();
+      responseBody.should.have.all.keys('components', 'job_queue_length', 'timestamp');
+      responseBody.job_queue_length.should.equal(loadBalancer.job_queue.length);
+      responseBody.timestamp.should.be.a('string');
+      responseBody.components.should.be.an('array');
+      responseBody.components.should.deep.include(lbTestStatus);
+      responseBody.components.forEach((elem, index) => {
+        if (index === 0 || index === 1 || index === 10) {
+          elem.status.should.equal('up');
+        } else {
+          elem.status.should.equal('down');
+        }
+      });
+      restoreConsole();
+      done();
+    });
+  });
+
+  it('when all ENs are down', (done) => {
+    stubConsole();
+    connectionCheckAssert([10], () => {
+      done();
+    });
+  });
+
+  /* it('when MySQL database is down', (done) => {
```

Comment:
If the code is not working, please remove the same from PR. It can always be added later. In the interim, the commented code can be retained in a separate branch on your fork.

Reply:
Do remember to remove the unnecessary tesring functions as well.